cli/azd/ci-build.ps1

```diff
@@ -39,43 +39,91 @@ if ($IsWindows) {
     Write-Host "go generate succeeded"
 }
 
+# Run `go help build` to obtain detailed information about `go build` flags.
+$buildFlags = @(
+    # Remove file system paths from the compiled binary
+    "-gcflags=-trimpath",
+    # Remove file system paths from the assembled code
+    "-asmflags=-trimpath",
+    # remove all file system paths from the resulting executable.
+    # Instead of absolute file system paths, the recorded file names
+    # will begin either a module path@version (when using modules),
+    # or a plain import path (when using the standard library, or GOPATH).
+    "-trimpath"
+)
+
+# Build constraint tags
+# cfi: Enable Control Flow Integrity (CFI),
+# cfg: Enable Control Flow Guard (CFG),
+# osusergo: Optimize for OS user accounts
+# netgo: Optimize for network I/O performance
+$tagsFlag = "-tags=cfi,cfg,osusergo,netgo"
+
+# ld linker flags
+# -s: Omit symbol table and debug information
+# -w: Omit DWARF symbol table
+# -X: Set variable at link time. Used to set the version in source.
+$ldFlag = "-ldflags=`"-s -w -X 'github.com/azure/azure-dev/cli/azd/internal.Version=$Version (commit $SourceVersion)' "
+
 if ($IsWindows) {
-    Write-Host "go build (windows)"
-    go build `
-        -buildmode=exe `
-        -tags="cfi,cfg,osusergo,netgo" `
-        -trimpath `
-        -gcflags="-trimpath" `
-        -asmflags="-trimpath" `
-        -ldflags="-s -w -X 'github.com/azure/azure-dev/cli/azd/internal.Version=$Version (commit $SourceVersion)' -linkmode=auto -extldflags=-Wl,--high-entropy-va"
+    Write-Host "Building for windows"
+    $buildFlags += @(
+        "-buildmode=exe",
+        $tagsFlag,
+        # -extldflags=-Wl,--high-entropy-va: Pass the high-entropy VA flag to the linker to enable high entropy virtual addresses
+        ($ldFlag + "-linkmode=auto -extldflags=-Wl,--high-entropy-va`"")
+    )
 }
 elseif ($IsLinux) {
-    Write-Host "go build (linux)"
-    go build `
-        -buildmode=pie `
-        -tags="cfi,cfg,cfgo,osusergo,netgo" `
-        -gcflags="-trimpath" `
-        -asmflags="-trimpath" `
-        -ldflags="-s -w -X 'github.com/azure/azure-dev/cli/azd/internal.Version=$Version (commit $SourceVersion)' -extldflags=-Wl,--high-entropy-va"
+    Write-Host "Building for linux"
+    $buildFlags += @(
+        "-buildmode=pie",
+        ($tagsFlag + ",cfgo"),
```

Comment:
@wbreza Couldn't find any immediate documentation on `cfgo`, let me know if there's something worth documenting here.

Reply:
I believe this is considered a custom build tag that I found as part of my initial research.  It's not required and can be removed.

Reply:
Making this a strict refactor. Pushing this out to #2250